pkg/kubectl/cmd/util/helpers.go

```diff
@@ -147,17 +147,29 @@ func checkErr(err error, handleErr func(string, int)) {
 		handleErr(MultilineError("Error in configuration: ", err), DefaultErrorExitCode)
 	default:
 		switch err := err.(type) {
-		case *meta.NoResourceMatchError:
+		case *NoResourceWithCommendError:
+			var msg string
+
 			switch {
 			case len(err.PartialResource.Group) > 0 && len(err.PartialResource.Version) > 0:
-				handleErr(fmt.Sprintf("the server doesn't have a resource type %q in group %q and version %q", err.PartialResource.Resource, err.PartialResource.Group, err.PartialResource.Version), DefaultErrorExitCode)
+				msg = fmt.Sprintf("the server doesn't have a resource type %q in group %q and version %q", err.PartialResource.Resource, err.PartialResource.Group, err.PartialResource.Version)
 			case len(err.PartialResource.Group) > 0:
-				handleErr(fmt.Sprintf("the server doesn't have a resource type %q in group %q", err.PartialResource.Resource, err.PartialResource.Group), DefaultErrorExitCode)
+				msg = fmt.Sprintf("the server doesn't have a resource type %q in group %q", err.PartialResource.Resource, err.PartialResource.Group)
 			case len(err.PartialResource.Version) > 0:
-				handleErr(fmt.Sprintf("the server doesn't have a resource type %q in version %q", err.PartialResource.Resource, err.PartialResource.Version), DefaultErrorExitCode)
+				msg = fmt.Sprintf("the server doesn't have a resource type %q in version %q", err.PartialResource.Resource, err.PartialResource.Version)
 			default:
-				handleErr(fmt.Sprintf("the server doesn't have a resource type %q", err.PartialResource.Resource), DefaultErrorExitCode)
+				msg = fmt.Sprintf("the server doesn't have a resource type %q", err.PartialResource.Resource)
+			}
+
+			if len(err.Recommends) != 0 {
+				msg += "\n\nDid you mean one of these?\n\n"
+				for _, recommend := range err.Recommends {
+					msg += fmt.Sprintf("\t%s\n", recommend)
```

Comment:
Are we indenting with tab elsewhere, or is this the only place?

Reply:
Thanks @smarterclayton for reviewing. 

The tabs are indenting for recommended commands. I think it would help. but I am open to options, like 2 spaces, 4 spaces, '- xxx', '* xxx' . What do you suggest?

```console
$ kubectl bet
Error: unknown command "bet" for "kubectl"

Did you mean this?
	set
	get

Run 'kubectl --help' for usage.
$ git bet
git: 'bet' is not a git command. See 'git --help'.

Did you mean this?
	bisect
```

Reply:
This is actually coming from cobra.